docs/05-technical-reference/00-architecture/svls-02-internal-registry.md

```diff
@@ -0,0 +1,17 @@
+---
+title: Internal Docker Registry
+---
+
+Internal docker registry is used for serverless to push built function images which can be used to deploy function without using 3rd party service.
+
+The following diagram illustrates how it works.
+
+![Serverless architecture](./assets/svls-internal-registry.svg)
+
+1. Build job pushes function image to docker registry using in-cluster URL
+2. Internal docker registry URL is resolved by K8s DNS to real ip address
+3. Kubelet fetch image using URL: `localhost:{node_port}/{image}`
```

Comment:
I think we should mention also the fact that when multiple nodes are used, the nodePort routing will assure that the request will reach the node where internal registry is deployed